### ISSUE TITLE
Folder Wizard Source Page has Broken Hardcoded Warning Color

### ISSUE THREAD

**Issue by elsiehupp** [nextcloud/desktop#3310]
## How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

## Expected Behavior
The warning color on the folder wizard source page should be darker when Dark Mode is enabled (and the text is set to white), in order to maintain legibility.

## Actual Behavior
The warning color on the folder wizard source page when Dark Mode is enabled (and the text is set to white) is identical to the warning color when Dark Mode is not enabled.

## Steps to Reproduce
On a version of macOS with Dark Mode enabled, open “Add Folder Sync Connection”.
> NOTE: This issue may also be present on Linux and/or Windows but has not been confirmed on either.

## Client Configuration
There are two examples posted in the comment thread on Issue https://github.com/nextcloud/desktop/issues/1287:

### Posted by @tmechen on [July 30, 2020](https://github.com/nextcloud/desktop/issues/1287#issuecomment-666388253):
* Client version: 2.6.5
* Operating system: macOS ????

![Bildschirmfoto 2020-07-30 um 20 31 11](https://user-images.githubusercontent.com/9206310/118066732-3a8c1c00-b36d-11eb-847b-518e44d8a3ec.png)

### Posted by @x00-R0n1n on [May 11, 2020](https://github.com/nextcloud/desktop/issues/1287#issuecomment-839219164):
* Client version: 3.2.1
* Operating system: macOS 10.15.7

![image](https://user-images.githubusercontent.com/22737601/117890794-a9cd1780-b26a-11eb-9bdc-c1a67f5ef0c5.png)

I was able to reproduce the behavior from the second image on:
* Client version: 3.2.1
* Operating system: macOS 11.3.1

As these two examples illustrate, the issue is not new and in fact seems to be compounding over time (with the background color of the parent widget—`FolderWizardSourcePage`—becoming light gray in the newer version of Nextcloud Desktop).

## Related Code

Using the text in the screenshots, I tracked down the likely source of the yellow color to this Qt `.ui` `<item>`:

https://github.com/nextcloud/desktop/blob/4efe00fecaf88b587c6bcb75f1af17ba1fa12747/src/gui/folderwizardsourcepage.ui#L53-L135

The portions with:
```xml
<color alpha="255"> 
  <red>255</red> 
  <green>255</green> 
  <blue>192</blue> 
</color> 
```
…seem to correspond with the light yellow in question, i.e. `#FFFFC0`.

A version of this yellow with an inverted luminosity would be:
```xml
<color alpha="255"> 
  <red>63</red> 
  <green>63</green> 
  <blue>0</blue> 
</color> 
```
…or `#3F3F00`.

I’m not sure why the background color of the parent widget (`FolderWizardSourcePage`) is light gray, but the cause is likely in either [`folderwizardsourcepage.ui`](https://github.com/nextcloud/desktop/blob/4efe00fecaf88b587c6bcb75f1af17ba1fa12747/src/gui/folderwizardsourcepage.ui) or [`folderwizard.cpp`](https://github.com/nextcloud/desktop/blob/00eca1260edddf9eb14d1312665de9842f6edbdc/src/gui/folderwizard.cpp).

**Comment by elsiehupp** [comment]
I just checked with the updated client:

* Client version: 3.3.3
* Operating system: macOS 11.5.2

<img width="616" alt="folder connection sheet" src="https://user-images.githubusercontent.com/9206310/132289782-5fceb0cb-3695-4b16-8963-0d0949bcf184.png">

I also came across [a potentially relevant tip](https://forum.qt.io/post/542991) on the Qt forums:

> [**khrykin**](https://forum.qt.io/user/khrykin) [Jul 25, 2019, 10:40 PM](https://forum.qt.io/post/542991)
>
> You can't reliably detect macOS appearance change in Qt, but nevertheless full dark mode support is achievable, though may require a bit of extra work. Key point is to do all your styling, i.e. backgrounds, borders, etc. in `paintEvent()` and relate all relevant colors to `QApplication::palette()`. Hence you can't really use `setStyleSheet()` for anything color-related. One particular gotcha is `QLabel`, for which I ended up writing my own `ColoredLabel` alternative which would receive its current color via callback.

I haven't looked particularly closely at the issue recently, but it seems like there are still some quirks in how Qt supports macOS Dark Mode.

**Comment by mgallien** [comment]
> I haven't looked particularly closely at the issue recently, but it seems like there are still some quirks in how Qt supports macOS Dark Mode.

My knowledge of macOS is not good enough. So is it easy to switch to dark mode ?
If yes, it means that we will be able to easily test and check if a fix is working.

@elsiehupp would you like to try working on this ? I guess you are impacted and we will make sure to find time to review your PR.

**Comment by elsiehupp** [comment]
> My knowledge of macOS is not good enough. So is it easy to switch to dark mode ?

Yes. It’s a simple toggle in System Preferences.

> would you like to try working on this ? I guess you are impacted and we will make sure to find time to review your PR.

I’m quite rusty on this codebase, and I’ve been focusing on other things, so I probably won’t get to it soon. This particular issue is just a tiny little thing that got spun out of a larger issue regarding Dark Mode compatibility. I came across the forum comment I linked to somewhat randomly, and it took some doing to dig up this issue and post an update. The forum comment I found may be applicable to https://github.com/nextcloud/desktop/issues/2994, as well, though.
